lib/internal/modules/esm/get_format.js

```diff
@@ -74,7 +74,7 @@ function getLegacyExtensionFormat(ext) {
 
 function getFileProtocolModuleFormat(url, ignoreErrors) {
   const ext = extname(url.pathname);
-  if (ext === '.js') {
+  if (ext === '.js' || !ext) {
```

Comment:
This has a broader impact than just "bin" files.

I could artificially limit it with something like [the parentDir check](https://github.com/nodejs/node/issues/33226#issuecomment-1001157586) I put together as a quick-fix to work around this issue for my own projects, so that it affects only "bin" files. I'm not sure whether the extra complexity would be worthwhile.

Is there a particular reason other extension-less files should specifically be excluded?

Reply:
What about if extensionless files are wasm in the future?

Reply:
Exclusively or additionally?

Reply:
Exclusive isn't possible; it'd have to be additionally.

Reply:
When more than 2 options are possible, I would probably just convert the logic to some kind of Map or Set.